File: 2-complexity/1-intro/main.cpp
// Introduction to Complexity - C++ Example

#include <iostream>
#include <vector>

int linear_search(const std::vector<int>& arr, int target) {
    for (size_t i = 0; i < arr.size(); i++) {
        if (arr[i] == target)
            return i; // return index
    }
    return -1; // not found
}

int main() {
    std::vector<int> data = {4, 7, 1, 3, 9};
    std::cout << "Index of 3: " << linear_search(data, 3) << std::endl;
    return 0;
}

File: 2-complexity/1-intro/main.java
// Introduction to Complexity - Java Example

class ComplexityIntro {

    public static int linearSearch(int[] arr, int target) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target) {
                return i;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] data = {4, 7, 1, 3, 9};
        System.out.println("Index of 3: " + linearSearch(data, 3));
    }
}
